src/main/java/com/limechain/network/protocol/blockannounce/messages/BlockAnnounceHandshakeBuilder.java

```diff
@@ -0,0 +1,39 @@
+package com.limechain.network.protocol.blockannounce.messages;
+
+import com.limechain.config.SystemInfo;
+import com.limechain.network.protocol.blockannounce.NodeRole;
+import com.limechain.rpc.server.AppBean;
+import com.limechain.storage.block.SyncState;
+import io.emeraldpay.polkaj.types.Hash256;
+
+import java.math.BigInteger;
+
+public class BlockAnnounceHandshakeBuilder {
+
+    /**
+     * Creates a Block Announce handshake based on the latest finalized Host state
+     *
+     * @return our Block Announce handshake
+     */
+    public BlockAnnounceHandshake getBlockAnnounceHandshake() {
+        SyncState syncState = AppBean.getBean(SyncState.class);
+        SystemInfo systemInfo = AppBean.getBean(SystemInfo.class);
+
+        Hash256 genesisBlockHash = syncState.getGenesisBlockHash();
+        Hash256 lastFinalizedBlockHash = syncState.getLastFinalizedBlockHash();
+        BigInteger lastFinalizedBlockNumber = syncState.getLastFinalizedBlockNumber();
+
+        NodeRole nodeRole = systemInfo.getNodeRole();
```

Comment:
We generally use `HostConfig` instead of `SystemInfo` for the purpose of fetching the node role. In the spirit of getting rid of `SystemInfo` as a bean, I'd change that around here, too.